src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,231 @@
+                PreparedStatement insertCarStatement
+                        = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            insertCarStatement.setString(1, car.getModel());
+            insertCarStatement.setLong(2, car.getManufacturer().getId());
+            insertCarStatement.executeUpdate();
+            ResultSet resultSet = insertCarStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't insert car " + car + " to DB", e);
+        }
+        addDriversToCar(car);
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT c.id, c.model, c.manufacturer_id, m.name, m.country "
+                + "FROM cars c "
+                + "JOIN manufacturers m "
+                + "ON c.manufacturer_id = m.id "
+                + "WHERE c.is_deleted = FALSE AND c.id = ?;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement selectCarStatement
+                        = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            selectCarStatement.setLong(1, id);
+            ResultSet resultSet = selectCarStatement.executeQuery();
+            if (resultSet.next()) {
+                car = getCarFromResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get car fro DB by car id " + id, e);
+        }
+        if (car != null) {
+            car.setDrivers(getDriversForCar(id));
+            return Optional.of(car);
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        List<Car> cars = new ArrayList<>();
+        String query = "SELECT c.id, c.model, c.manufacturer_id, m.name, m.country "
+                + "FROM cars c "
+                + "JOIN manufacturers m "
+                + "ON c.manufacturer_id = m.id "
+                + "WHERE c.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                Statement getAllCarsStatement = connection.createStatement()) {
+            ResultSet resultSet = getAllCarsStatement
+                    .executeQuery(query);
+            while (resultSet.next()) {
+                Car car = getCarFromResultSet(resultSet);
+                cars.add(car);
+            }
+
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get all cars from DB", e);
+        }
+        if (cars.size() > 0) {
+            for (Car car: cars) {
+                car.setDrivers(getDriversForCar(car.getId()));
+            }
+        }
```

Comment:
```suggestion
            for (Car car: cars) {
                car.setDrivers(getDriversForCar(car.getId()));
            }
```

why do we need this check?